prediction_market_agent/agents/safe_guard_agent/guards/agent.py

```diff
@@ -11,33 +14,44 @@
 )
 
 
-@observe()
-def validate_do_not_remove_agent(
-    new_transaction: DetailedTransactionResponse,
-    new_transaction_safetx: SafeTx,
-    all_addresses_from_tx: list[ChecksumAddress],
-    history: list[DetailedTransactionResponse],
-) -> ValidationResult:
-    ok_result = ValidationResult(
-        ok=True, reason="Doesn't remove agent from the owner list."
-    )
-
-    if not new_transaction.txData:
-        return ok_result
-
-    if not new_transaction.txData.dataDecoded:
-        return ok_result
-
-    if new_transaction.txData.dataDecoded.get("method") != "removeOwner":
-        return ok_result
-
-    if (
-        # Based on https://github.com/safe-global/safe-smart-account/blob/main/contracts/base/OwnerManager.sol#L73.
-        new_transaction.txData.dataDecoded["parameters"][1]["value"].lower()
-        != APIKeys().bet_from_address.lower()
-    ):
-        return ok_result
-
-    return ValidationResult(
-        ok=False, reason="Agent can not accept removal of itself from the owners."
-    )
+class DoNotRemoveAgent(AbstractGuard):
+    name = "Agent remains owner"
+    description = "This guard ensures that the transaction doesn't remove the agent itself from the owners of the Safe."
+
+    @observe(name="validate_do_not_remove_agent")
+    def validate(
+        self,
+        new_transaction: DetailedTransactionResponse,
+        new_transaction_safetx: SafeTx,
+        all_addresses_from_tx: list[ChecksumAddress],
+        history: list[DetailedTransactionResponse],
+    ) -> ValidationResult:
+        ok_result = ValidationResult(
+            name=self.name,
+            description=self.description,
+            ok=True,
+            reason="The transaction does not remove the agent from owners.",
+        )
+
+        if not new_transaction.txData:
+            return ok_result
+
+        if not new_transaction.txData.dataDecoded:
+            return ok_result
```

Comment:
Probably a silly question - any reason for combining `not a or not b or ...`?

Reply:
Not really, I'm fine with that 😄